lib/tasks/convert_embeddables.rake

```diff
@@ -205,98 +176,131 @@ def convert()
+        embeddable.destroy
+        deleted_embeddables += 1
+      end
+    end
+  end
+  puts "Deleted #{deleted_embeddables} orphaned old embeddables."
 end
 
 def replace_old_embeddables()
-  # Hook up any ManagedInteractive embeddables that have a legacy_ref_id value and are not 
-  # associated with a PageItem. Using the ManagedInteractive embeddable's legacy_ref_id,
-  # find the built-in embeddable and PageItem that it is associated with. Then, replace the 
-  # PageItem's built-in embeddable with the ManagedInteractive embeddable. (Delete the old, 
-  # built-in embeddable?) We will also need to add LinkedPageItems to
-  ManagedInteractive.where("legacy_ref_id IS NOT NULL AND legacy_ref_id != ?", "").each do |mi|
-    if PageItem.where(:embeddable_id => mi.id, :embeddable_type => "ManagedInteractive").count == 0
-      old_embeddable_id, old_embeddable_type = mi.legacy_ref_id.split("-")
-      page_item = PageItem.where(:embeddable_id => old_embeddable_id, :embeddable_type => old_embeddable_type).first
-      if page_item
-        puts "page_item: #{page_item.inspect}"
-        # TODO: Update the page item's embeddable to be the ManagedInteractive embeddable.
+  # To replace the old, built-in embeddables, first we find any ManagedInteractive embeddables
+  # that have a legacy_ref_id value and are not yet associated with a PageItem. Then, we use 
+  # each of these ManagedInteractive embeddables' legacy_ref_id to find the old embeddable it 
+  # will replace and the PageItem that it's associated with. We replace the PageItem's old 
+  # embeddable with the new embeddable. We also update any wrapping plugins for the old embeddable
+  # so they point to the new embeddable. Finally, delete the old embeddable.
+  new_embeddables = ManagedInteractive.where("legacy_ref_id IS NOT NULL AND legacy_ref_id != ?", "")
+  new_embeddables.each do |new_embeddable|
+    if PageItem.where(:embeddable_id => new_embeddable.id, :embeddable_type => "ManagedInteractive").count == 0
+      old_embeddable_id, old_embeddable_type = new_embeddable.legacy_ref_id.split("-")
+      old_embeddable = old_embeddable_type.constantize.find(old_embeddable_id)
+      page_items = PageItem.where(:embeddable_id => old_embeddable.id, :embeddable_type => old_embeddable_type)
+      page_items.each do |page_item|
+        page_item.embeddable = new_embeddable
+        page_item.save!
+        wrapping_plugins = Embeddable::EmbeddablePlugin.where(:embeddable_id => old_embeddable.id, :embeddable_type => old_embeddable_type)
+        wrapping_plugins.each do |wrapping_plugin|
+          wrapping_plugin.embeddable = new_embeddable
+          wrapping_plugin.save!
+        end
+        old_embeddable.destroy
       end
     end
   end
-  # TODO: Make sure TE question wrappers are updated to point at new ManagedInteractive.
-
+  # TODO: Set associated activities' runtime to "Activity Player" after all old embeddables are replaced?
```

Comment:
Yes and also update the migration status field. You might want to update this field to indicate the conversion of the activity is in progress and then update it again when the activity has been fully converted.